### PR TITLE
CONSOLE-2444: deployment i18n

### PR DESCRIPTION
**Pull request by zherman0**
[CONSOLE-2444](https://issues.redhat.com/browse/CONSOLE-2444): deployment i18n

Needs PR7043, PR7056, and PR7025 to be complete.

### CODE REVIEW

File: frontend/public/components/deployment.tsx [hunk 1]
```diff
@@ -79,53 +84,63 @@ export const menuActions = [
...
           <DetailsItem
-            label="Max Unavailable"
+            label={t('workload~Max unavailable')}
             obj={deployment}
             path="spec.strategy.rollingUpdate.maxUnavailable"
           >
-            {deployment.spec.strategy.rollingUpdate.maxUnavailable || 1} of{' '}
-            {pluralize(deployment.spec.replicas, 'pod')}
+            {deployment.spec.strategy.rollingUpdate.maxUnavailable || 1} {t('workload~of ')}
+            {t('workload~pod', { count: deployment.spec.replicas })}
```
Comment (spadgett) [inline frontend/public/components/deployment.tsx:103]: This should be one message with multiple parameters

File: frontend/public/components/deployment.tsx [hunk 2]
```diff
@@ -79,53 +84,63 @@ export const menuActions = [
...
           <DetailsItem
-            label="Max Surge"
+            label={t('workload~Max surge')}
             obj={deployment}
             path="spec.strategy.rollingUpdate.maxSurge"
           >
-            {deployment.spec.strategy.rollingUpdate.maxSurge || 1} greater than{' '}
-            {pluralize(deployment.spec.replicas, 'pod')}
+            {deployment.spec.strategy.rollingUpdate.maxSurge || 1} {t('workload~greater than ')}
+            {t('workload~pod', { count: deployment.spec.replicas })}
```
Comment (spadgett) [inline frontend/public/components/deployment.tsx:111]: This should be one message

**Review by spadgett — APPROVED** [review]
/lgtm
/hold for approvals
/assign @yapei @ahardin-rh @sferich888 

Let's open a separate BZ for the two comments I have so we can backport a fix.

**Comment by spadgett** [comment]
I opened https://bugzilla.redhat.com/show_bug.cgi?id=1894110 and #7083 for the max unavailable / max surge bug.

**Comment by yanpzhan** [comment]
@zherman0 Could you rebase your pr? I'd like to have a test against your pr based on latest code.

**Comment by zherman0** [comment]
@yanpzhan @spadgett - Rebased this with the cherry picked changes. Ready for review.

**Review by spadgett — APPROVED** [review]
/lgtm
/hold cancel

**Comment by spadgett** [comment]
/lgtm cancel

Looks like we have test failures.

> [SEVERE] https://console-openshift-console.apps.ci-op-9j5mq7qi-75d12.origin-ci-int-gce.dev.openshift.com/static/main-chunk-a651199ab34fd4a7d03e.min.js 0:386439 "Missing i18n key \"{{maxSurge}} greater than pod_plural\" in namespace \"workload\" and language \"en.\""

**Comment by zherman0** [comment]
@spadgett - super weird, somehow lost that key on the rebase.  Fixed it now. Actually, probably happened due to conflict with cherry picked fix with ?? vs ||.

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by zherman0** [comment]
rebased. @spadgett please add back lgtm.

**Review by spadgett — APPROVED** [review]
/lgtm

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**